mempool/src/tests/core_mempool_test.rs

```diff
@@ -672,3 +696,20 @@ fn test_get_transaction_by_hash_after_the_txn_is_updated() {
     let txn_by_new_hash = pool.get_by_hash(new_txn_hash);
     assert_eq!(txn_by_new_hash, Some(new_txn));
 }
+
+#[test]
+fn test_bytes_limit() {
+    let mut config = NodeConfig::random();
+    config.mempool.capacity = 100;
+    let mut pool = CoreMempool::new(&config);
+    // add 100 transacionts
+    for seq in 0..100 {
+        add_txn(&mut pool, TestTransaction::new(1, seq, 1)).unwrap();
+    }
+    let get_all = pool.get_batch(100, 100 * 1024, HashSet::new());
+    assert_eq!(get_all.len(), 100);
+    let txn_size = get_all[0].raw_txn_bytes_len() as u64;
+    let limit = 10;
+    let hit_limit = pool.get_batch(100, txn_size * limit, HashSet::new());
+    assert_eq!(hit_limit.len(), limit as usize);
```

Comment:
not really testing the bytes limit being hit?

Reply:
it only returns 10 txns instead of 100?

Reply:
ugh. for sure. I'm stupid.